### ISSUE TITLE
PCP: `sql_def` only used in some pages

### ISSUE THREAD

**Issue by vendethiel** [IntegraMOD/IntegraMOD151#47]
Hey,

I'm trying to find a way to do what @MWE001 asked for.

So I went in classes and changed the `webdisplay`'s "SQL Definition" (`sql_def`) to 0, just to try to cancel it out, and hide it.

But... It seems like only `profilcp/profilcp_buddy` uses the `sql_def` :o.
Am I misunderstanding what `sql_def` is about? Is this a PCP "gotcha" I need to understand?
Is it just some missing code to deal with it?

**Comment by vendethiel** [comment]
Let's just say I can't read! 

This is on the page:

> This the sql definition for this class used in the buddy/members list.	

Whoops. My bad.


Okay. Let's move on then!

@MWE001 (and others...) Should we (or I, really, if needed) add a "filter" of some sorts?
It's already possible in fields>Input or Output. But maybe we want to add a regexp, or some SQL to make sure the user is allowed to update that field (or it's displayable), or something else. Not sure?

**Comment by IntegraMOD** [comment]
I just messaged Teelk. He was one of our devs with a firm grasp of PCP and really anything Integramod. If he responds im sure he can answer any questions.

**Comment by vendethiel** [comment]
That'd be really amazing. Thank you very much :-).

**Comment by vendethiel** [comment]
So I just went and downloaded a "vanilla" PCP.

This still leaves a question : PCP, in usermaps (say we go in the webdisplay usermap) has a "Auth Level". But it's in **input**? And worse. Trying to grep for `'auth'` in `profilecp/` does find the *definitions* of auth for each field... But I see no piece of code to check for it O.ô.

Resting my case for now...

Note: as I discovered... The `pcp_output` (contents) has been "removed" by PCP Extra to have it all in `display_field`.

**Comment by MWE001** [comment]
What did I ask for? I'm scatter brained. I think it was the no website in profile for now?

And speaking of PCP, I have a file for for you. Not going to side track here so Ill PM you.

**Comment by vendethiel** [comment]
hey @MWE001 , did you end up sending the file to me somewhere? :)

**Comment by MWE001** [comment]
What file would that be? The adr PCP files?

Ray

**Comment by vendethiel** [comment]
I'm not sure. You said this:

> And speaking of PCP, I have a file for for you. Not going to side track here so Ill PM you.

**Comment by MWE001** [comment]
Ok. Yes, that would be the files to integrate ad&r into pcp.

I can put them in my Dropbox if you want them.

**Comment by vendethiel** [comment]
Either that or on im.com on some topic :).

**Comment by MWE001** [comment]
Ok. Give me about 10-15 minutes to get in front of my PC.

**Comment by MWE001** [comment]
Ok buddy. Posted at integra for you.

**Comment by vendethiel** [comment]
Ok, catching up after all the real-life stuff happened, this one is good to close. It's working as intended!

**Comment by vendethiel** [comment]
Gonna reopen for something a bit different.

I was experiencing a very strange bug with PCP.
I wanted to hide the message "(visible by administrators only)" under the "Password" field, because it looks like the administrators can see your password, which is bad.

So, I went to the `Root » PCP » profil » profile_prefer » Registering_info` map, edited `user_password`, changed "Visibility" to false.
Took me a few minutes to understand I had to submit the map as well, after submitting the field.

So, I submit both, refresh the registration page... And nothing changed.
Go back to the map, and it says the field is back to Visibility > ON.

After an hour or so of debugging, I finally understood what happened:
- If you have a field with a setting to "ON"
- You try to override the value to "OFF" in a specific map
- The setting doesn't save

Took me some time to figure out, that the code checked for "empty" values when writing the settings, and that "OFF" is considered "empty"...
It seems like that's been broken for 15 years now. Oops.

I also uncovered a CTracker issue in the userdef fields, because of course...
